flash/core/data/utilities/paths.py

```diff
@@ -152,18 +153,37 @@ def filter_valid_files(
     if not isinstance(files, List):
         files = [files]
 
+    if valid_extensions is None:
+        return (files,) + additional_lists
+
+    if not isinstance(valid_extensions, tuple):
+        valid_extensions = tuple(valid_extensions)
+
     additional_lists = tuple([a] if not isinstance(a, List) else a for a in additional_lists)
 
     if not all(len(a) == len(files) for a in additional_lists):
         raise MisconfigurationException(
             f"The number of files ({len(files)}) and the number of items in any additional lists must be the same."
         )
 
-    if valid_extensions is None:
-        return (files,) + additional_lists
     filtered = list(
         filter(lambda sample: has_file_allowed_extension(sample[0], valid_extensions), zip(files, *additional_lists))
     )
+
     if len(additional_lists) > 0:
         return tuple(zip(*filtered))
```

Comment:
We should have the warning for this case too. Can just move the if to the end of the function, but will need to override the `filtered` variable and give it a different name instead like `filtered_files` or something.

Reply:
sanity checking:

if additional_lists exists, the files in additional lists should be removed from filtered before being returned?

regarding the warning: do we just want to tell the user only the filtered files originating from the files param are being returned?

Reply:
I also added a test for `additional_lists` in that last commit

Reply:
once the PR is merged; do you mind if I open another issue + PR to explicitly handle tests for the remainder of paths.py (I'd like to have a PR on record for testing specifically)

Reply:
Awesome!

> once this PR is merged; do you mind if I open another issue + PR to explicitly handle tests for the remainder of paths.py (I'd like to have a PR on record for testing specifically)

Go for it! Definitely an area that could do with thorough testing :smiley: